test/e2e/create_cluster_test.go

```diff
@@ -114,13 +114,16 @@ func TestCreateClusterCustomConfig(t *testing.T) {
 	err = client.Get(testContext, crclient.ObjectKeyFromObject(hostedCluster), hostedCluster)
 	g.Expect(err).NotTo(HaveOccurred(), "failed to get hostedcluster")
 
-	e2eutil.EnsureNoCrashingPods(t, ctx, client, hostedCluster)
+	e2eutil.EnsureNoCrashingPodsInHCP(t, ctx, client, hostedCluster)
 	e2eutil.EnsureNodeCommunication(t, ctx, client, hostedCluster)
 
 	e2eutil.EnsureSecretEncryptedUsingKMS(t, ctx, hostedCluster, guestClient)
 
 	// test oauth with identity provider
 	e2eutil.EnsureOAuthWithIdentityProvider(t, ctx, client, hostedCluster)
+
+	// test the cluster behind a proxy.
```

Comment:
This seems to validate changing proxy day 2. I'd still like to have an e2e like current `TestCreateClusterProxy` where we validate that creation succeed when the proxy config is set.